docs/source/whitepaper.rst

```diff
@@ -335,20 +334,82 @@ Charles can attempt to delay Bob's withdrawal by challenging Bob's rightful clai
 is sufficient to cover Bob's fee for L1 resolution, Bob will proceed with L1 resolution. If not, Bob can continue opening
 parallel claims until Charles no longer contests one of them, or there is enough accumulated stake from Charles on the
 multiple challenges for Bob to do an L1 resolution. In any case, Bob will be able to prove his rightful claim before
-Charles' claim reach the end of its period.
-The time constraint described in the previous example also holds in this case.
+Charles' claim reach the end of its period. The time constraint described in the previous example also holds in this case.
 
 Claims that cannot be filled
 ++++++++++++++++++++++++++++
 
 In both the regular `false claim` and `self-challenge claim` cases, we assumed that Bob could fill Alice's request in
 order to prove that the false claimer Charles was not the correct filler. However, If Alice's request cannot be filled
 for any reason (e.g. transfer value too high), instead of proving that someone other than Charles filled a request,
-Bob will need to prove that no one filled the request before a certain block height. For that, Bob needs to create and
-submit an `L1 non-fill proof` from rollup B to rollup A.
+Bob will need to prove that Charles did not fill the request as claimed. For that, Bob needs to create and submit an
+`L1 non-fill proof` from rollup B to rollup A.
 
-.. todo::
-    Exact specification TBD: https://github.com/beamer-bridge/beamer/issues/346
+When called, the fill manager contract on rollup B recomputes the fill hash from the request hash, and fill ID,
+and checks that no fills exists for the corresponding request hash and fill hash. It then submits a proof
+to the outbox of rollup B indicating that the fill hash is invalid, i.e. that the request hash cannot be mapped to the
+fill hash.
+
+Similarly to the filled L1 resolution case, Bob can then trigger a call on L1 to forward this message to rollup A. This
+message will store in the `resolution registry` that the `fill hash` is invalid and make any claim from Charles with
+the corresponding hash an invalid claim.
+
+To make sure the proof arrives in time on rollup A, Bob will need to call the `fill manager` as soon as he notices a
+false claim for a non-filled request. It takes `finalization time of rollup B` after Bob's call to be able to send the
+proof, while the challenge period is defined to be `finalization time of rollup B + challenge period extension`.
+
+In the case where someone challenges Charles on the false claim at the same time as Bob sends the transaction for the
+proof on rollup B, Bob will not be able to challenge Charles and will not receive any financial reward from having sent
```

Comment:
Maybe it makes sense to rewrite this in the future because I think we should explain the new challenge mechanism when claims are explained. If the reader is aware of how challenge games work at this point, we can rewrite it a bit that he might not be financially rewarded if he is not part of the challenge at any point of the challenge.